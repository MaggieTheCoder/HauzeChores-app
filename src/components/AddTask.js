import { React, useState } from "react";
import "../styles/AddTask.css";

const AddTask = () => {
  const [fields, setFields] = useState();
  const [custom, setCustom] = useState(false);

  const handleAddTask = (event) => {
    event.preventDefault();
    // console.log("****", fields);
  };
  const handleFieldChange = (event) => {
    console.log("**", event.target.value);
    if (event.target.value === "custom") {
      setCustom(true);
    } else {
      setCustom(false);
      setFields(event.target.value);
    }
  };
  const handleInput = (event) => {
    // console.log(event.target.value, fields);
    setFields(event.target.value);
  };

  return (
    <>
      <div className="add-task">
        <form className="task__list" onSubmit={handleAddTask}>
          <label htmlFor="type">
            Type
            <select
              id="type"
              name="type"
              value={fields}
              onChange={handleFieldChange}
            >
              <option value="Empty Dishwasher">Empty Dishwasher</option>
              <option value="Hoover living room">Hoover living room</option>
              <option value="Water the plants">Water the plants</option>
              <option value="custom">Add your own</option>
            </select>
          </label>

          {!custom && <button type="submit">Add Task</button>}
        </form>
      </div>
      {custom && (
        <div className="custom-task">
          <form onSubmit={handleAddTask}>
            <input type="text" onChange={handleInput} placeholder="add task" />
            <button type="submit">Add Task</button>
          </form>
        </div>
      )}
    </>
  );
};

export default AddTask;
