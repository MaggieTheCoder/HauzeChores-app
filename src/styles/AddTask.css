
.add-task {
  padding-top: 10px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  background-color: #99E5FF;
  height: 120px;

  
}
label {
  font-family: "Menlo", sans-serif;
  font-size: 25px;
  color: #C60982;
  margin-right: 20px;
}

.task__dropdown {
  font-family: "Menlo", sans-serif;
  font-size: 20px;
  font-style: italic;
  margin: 10px;
  height:50px;
  border-radius: 10px;
  padding: 0px 60px 0px 20px;
}
.add_task_button {
  font-family: "Menlo", sans-serif;
  font-size: 25px;
  background-color: #C60982;
  padding: 0px 50px 0px 50px;
  border-radius: 15px;
  color: white;
  height:50px;
  
}

.custom-task {
  background-color: #99E5FF;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  height: 80px;
  padding-bottom: 100px;
}
.add_task_button_custom {
  font-family: "Menlo", sans-serif;
  font-size: 25px;
  background-color: #C60982;
  padding: 0px 60px 0px 60px;
  border-radius: 15px;
  color: white;
  height:50px;
  margin-left: 20px;
}

.custom-task-input {
  font-family: "Menlo", sans-serif;
  font-style: italic;
  font-size: 20px;
  margin: 10px;
  height:50px;
  border-radius: 10px;
  height:50px;
  padding: 0px 90px 0px 20px;
}
