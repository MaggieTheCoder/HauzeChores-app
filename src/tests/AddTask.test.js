import React from "react";
import { render, screen } from "@testing-library/react";
import AddTask from "../components/AddTask";

describe("AddTask component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<AddTask />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("button renders with correct text", () => {
    render(<AddTask />);
    const buttonElement = screen.getByText("Add Task");
    expect(buttonElement).toBeInTheDocument();
  });
});
