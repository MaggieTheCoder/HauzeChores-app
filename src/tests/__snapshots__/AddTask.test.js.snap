// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-task"
  >
    <form
      class="task__list"
    >
      <label
        for="type"
      >
        Type
        <select
          id="type"
          name="type"
        >
          <option
            value="Empty Dishwasher"
          >
            Empty Dishwasher
          </option>
          <option
            value="Hoover living room"
          >
            Hoover living room
          </option>
          <option
            value="Water the plants"
          >
            Water the plants
          </option>
        </select>
      </label>
      <button
        type="submit"
      >
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
